ddtrace/llmobs/_writer.py

```diff
@@ -107,6 +106,13 @@ def periodic(self) -> None:
             events = self._buffer
             self._buffer = []
 
+        if not self._headers.get("DD-API-KEY"):
```

Comment:
Is this a fix? Not sure how this relates to this PR - if it's not necessary would love to see it be in a smaller separate PR

Reply:
The existing behavior is that we log a warning if an api key isn't found inside `submit_evaluation`. For `submit_evaluation_for`, we change the behavior to warning log before we try to submit instead if an api key is missing

See:
https://github.com/DataDog/dd-trace-py/pull/11535#discussion_r1857318812
https://github.com/DataDog/dd-trace-py/pull/11535#discussion_r1859060367